From fcad550f0452963eecd76a445a4ebea424d0fa8b Mon Sep 17 00:00:00 2001
From: kboeluek <kboeluek@2eb42a87-3827-0410-b767-dc15055982e4>
Date: Tue, 23 Dec 2014 14:23:53 +0000
Subject: [PATCH 172/462] MANTIS 0035039: MATAC Client Added the MatacClient to
 the list of projects to build. An extra entry in AntProperties.xml was needed
 to reference the new jar. An extra target was added to AntTargets.xml to
 automatically generate code from a WSDL file

git-svn-id: svn://localhost/SelfServiceBuild/trunk@255 2eb42a87-3827-0410-b767-dc15055982e4
---
 SelfServiceBuild/Build/java/AntProperties.xml |  1 +
 SelfServiceBuild/Build/java/AntTargets.xml    | 19 +++++++++++++++++++
 SelfServiceBuild/Build/java/build.xml         |  2 +-
 3 files changed, 21 insertions(+), 1 deletion(-)

diff --git a/SelfServiceBuild/Build/java/AntProperties.xml b/SelfServiceBuild/Build/java/AntProperties.xml
index 20486a4..7ec2aa4 100644
--- a/SelfServiceBuild/Build/java/AntProperties.xml
+++ b/SelfServiceBuild/Build/java/AntProperties.xml
@@ -224,6 +224,7 @@
 <property name="JarMassaiCORBASettings"     value="${JarDir}/massai-corba-settings.jar"/>
 <property name="JarMassaiReleaseNotes"      value="${JarDir}/releasenotes.jar"/>
 <property name="JarMassaiStatechart"        value="${JarDir}/massai-statechart.jar"/>
+<property name="JarMatacClient"             value="${JarDir}/MATAC-CLIENT.jar"/>
 
 <!-- Definition des IDL-Compilers: OpenORB ======================= -->
 
diff --git a/SelfServiceBuild/Build/java/AntTargets.xml b/SelfServiceBuild/Build/java/AntTargets.xml
index 11563b1..29f8c8e 100644
--- a/SelfServiceBuild/Build/java/AntTargets.xml
+++ b/SelfServiceBuild/Build/java/AntTargets.xml
@@ -120,6 +120,24 @@
     <foreach list="${project.xmlBeanDir}" delimiter=" " param="SchemaDir" target="create-xml-bean-dir" inheritall="true"/>
 </target>
 
+<!-- create-from-wsdl ================================================= -->
+
+<target name="create-from-wsdl" if="project.wsdlFile">
+    <mkdir dir="${GenSrcDir}"/>
+    <java classname="org.apache.axis2.wsdl.WSDL2Java" fork="true">
+      <classpath refid="axis2.classpath"/>
+      <arg value="-uri"/>
+      <arg file="${project.wsdlFile}"/>
+      <arg value="-o"/>
+      <arg value="${GenSrcDir}"/>
+      <arg value="-p"/>
+      <arg value="-d"/>
+      <arg value="xmlbeans"/>
+      <arg value="-s"/>
+      <arg value="--noBuildXML"/>
+    </java>
+    <jar basedir="${GenSrcDir}/resources" destfile="${JarDir}/${xbeans.packaged.jar.name}.jar" excludes="**/services.xml, **/*.xsd"/>
+</target>
 
 <!-- prepare-compile ============================================= -->
 
@@ -187,6 +205,7 @@
         depends="create-flex,
                  create-xml-bean,
                  create-xml-bean-directories,
+                 create-from-wsdl,
                  create-idl,
                  prepare-compile,
                  compile-src">
diff --git a/SelfServiceBuild/Build/java/build.xml b/SelfServiceBuild/Build/java/build.xml
index 5f54d9d..ef3e8b8 100644
--- a/SelfServiceBuild/Build/java/build.xml
+++ b/SelfServiceBuild/Build/java/build.xml
@@ -24,7 +24,7 @@
 
     <!-- Definition der Verzeichnisse mit den Java-Modulen ======= -->
     <property name="CommonDirList"   value="${CommonJavaDir}/OpenORB-1.3.1/OpenORB"/>
-    <property name="PlatformDirList" value="${PlatformJavaDir}/CUSS,${PlatformJavaDir}/OPI.1.3,${PlatformJavaDir}/Tools,${PlatformJavaDir}/CussIF,${PlatformJavaDir}/Config,${PlatformJavaDir}/icnTunnelClient,${PlatformJavaDir}/PlatformManager,${PlatformJavaDir}/PrintService,${PlatformJavaDir}/CORBA-Settings,${PlatformJavaDir}/CLAApplet"/>
+    <property name="PlatformDirList" value="${PlatformJavaDir}/CUSS,${PlatformJavaDir}/OPI.1.3,${PlatformJavaDir}/Tools,${PlatformJavaDir}/MATAC-Client,${PlatformJavaDir}/CussIF,${PlatformJavaDir}/Config,${PlatformJavaDir}/icnTunnelClient,${PlatformJavaDir}/PlatformManager,${PlatformJavaDir}/PrintService,${PlatformJavaDir}/CORBA-Settings,${PlatformJavaDir}/CLAApplet"/>
     <property name="AblsDirList"     value="${AblsJavaDir}/KioskProxy,${AblsJavaDir}/TunnelServer"/>
     <property name="AppDirList"      value=""/>
 
-- 
2.41.0.windows.1

